malloc_count.c

```diff
@@ -122,6 +125,30 @@ extern void malloc_count_reset_peak(void)
     peak = curr;
 }
 
+/* user function to return total allocation volume */
+extern size_t malloc_count_total(void)
+{
+    return local_total;
+}
+
+/* user function to reset total allocation volume to 0 */
+extern void malloc_count_reset_total(void)
+{
+    local_total = 0;
```

Comment:
`malloc_count_reset_peak` also introduces global state, I fail to see the difference. I can change `local_total` to `last_total` and return the difference in `malloc_count_total` if that's what you mean.